examples/demo.go

```diff
@@ -10,53 +10,54 @@ import (
 )
 
 func main() {
-	distro := wsl.Distro{Name: "Ubuntu-22.04-test"}
+	distro := wsl.Distro{Name: "Ubuntu-GoWSL-demo"}
 
 	// Registering a new distro
-	// TODO: install ubuntu-preview from the store?
-	fmt.Println("Registering a new WSL distro...")
-	if err := distro.Register(`.\images\jammy.tar.gz`); err != nil {
+	fmt.Printf("Registering a new distro %q\n", distro.Name)
+	if err := distro.Register(`.\images\rootfs.tar.gz`); err != nil {
 		panic(err)
 	}
 
 	// Ensuring the distro is unregistered at the end
 	defer distro.Unregister()
 
 	// Getting config and printing it
+	fmt.Println("\nPrinting distro information:")
 	fmt.Printf("%v", distro)
 
 	// Setting config
+	fmt.Println("\nDisable windows paths and fail to run notepad:")
 	distro.PathAppended(false)
 
-	// Launching async command
-	process1 := distro.Command(context.Background(), `sleep 3 && cat goodmorning.txt`)
-	process1.Start()
-
-	// Launching async command
-	process2 := distro.Command(context.Background(), `echo "Hello, world from WSL!" > "goodmorning.txt"`)
-	process2.Run()
-
 	// Launching an interactive command (should fail as per config change)
-	err := distro.Shell(wsl.WithCommand("sh -c 'notepad.exe'"))
+	out, err := distro.Command(context.Background(), "sh -c 'notepad.exe'").CombinedOutput()
 	if err != nil {
-		fmt.Printf("Sync command unsuccesful: %v\n", err)
+		fmt.Printf("%v\n%s", err, out)
 	} else {
-		fmt.Printf("Sync command succesful\n")
+		fmt.Printf("Unexpected success:\n%s", out)
 	}
 
-	// Managing the result of the async commands
-	err = process1.Wait()
+	// Launching async command 1
+	fmt.Println("\nRunning async commands")
+	cmd1 := distro.Command(context.Background(), `sleep 3 && cat goodmorning.txt`)
+	cmd1.Start()
+
+	// Launching and waiting for command 2
+	distro.Command(context.Background(), `echo "Hello, world from WSL!" > "goodmorning.txt"`).Run()
+
+	// Waiting for command 1
+	err = cmd1.Wait()
 	if err != nil && !errors.Is(err, &wsl.ExitError{}) {
```

Comment:
Yeah, I know, this is not part of the diff, but while you are cleaning this up, I would nest this time the errors.is check.

Basically, you want to check "was there an error?", and then depending on the type of error, either panic or print + return.